tests/event-processing-tests/src/test/java/uk/gov/di/txma/eventprocessor/bdd/LambdaToS3StepDefinitions.java

```diff
@@ -384,4 +424,34 @@ private static int findBracket(String input, int start) {
         }
         return 0;
     }
+
+    /**
+     * This compares a message in S3 to the expected S3 output
+     *
+     * @param S3            The S3 message to be compared
+     * @param expectedS3    The expected message
+     * @return              True or false depending on if the S3 message contains the expected S3
+     */
+    private boolean compareOutput(JSONObject S3, JSONObject expectedS3){
+
+        // Loops through the keys of the expected result
+        Iterator<String> keys = expectedS3.keys();
+        while(keys.hasNext()) {
```

Comment:
Not a big deal since this is test code, but going forward better to use a lambda and the forEachRemaining method eg.
```
keys.forEachRemaining(key -> {
if (S3.has(key)) {...}
...
})
```